src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -738,6 +740,39 @@ function validate_connection_args(user_input, action) {
     }
 }
 
+////////////////////////////
+//// UPGRADE VALIDATION ////
+///////////////////////////
+
+/**
+ * validate_expected_version checks that expected_version is valid
+ * if it's not valid it throws an error
+ * @param {string} expected_version
+ */
+function validate_expected_version(expected_version) {
+    if (!expected_version) {
+        throw new Error('expected_version flag is required');
+    }
+    if (!version_utils.is_valid_sematic_version(expected_version)) {
+        throw new Error('expected_version must have sematic version structure (major.minor.patch)');
+    }
+    if (check_if_expected_version_newer_that_package_version(expected_version)) {
+        throw new Error(`expected_version cannot be later that the package version ${pkg.version}`);
+    }
+}
+
+/**
+ * compare_version_to_current_version checks that the version
+ * is not later that the one that was defined in the pkg
+ * @param {string} expected_version
+ * @returns {boolean}
+ */
+function check_if_expected_version_newer_that_package_version(expected_version) {
```

Comment:
can we move this method also to new util file?

Reply:
It's about an argument, the `versions_utils.js` is more general (like a library that we can use).